src/primitives-scenegraph/PrimitivesSceneGraphExample.cpp

```diff
@@ -0,0 +1,209 @@
+        Matrix4::translation(Vector3::zAxis(-10.0f));
+
+	/* Set up objects*/
+	(*(_objects[0] = new DrawableObject{_shader, 0x3bd267_rgbf, _cube, _scene, _drawables }))
+		.rotateZ(45.0_degf)
+		.rotateX(-45.0_degf)
+		.translate({ 0.5f, 0.0f, -2.0f });
+
+	(*(_objects[1] = new DrawableObject{_shader, 0x2f83cc_rgbf, _sphere, _scene, _drawables }))
+		.scale(Vector3(0.75f))
+		.translate({ -1.2f, -0.3f, -0.2f });
+
+	/* This uvSphereWireframe object has its parent set to the uvSphereSolid object, rather than
+		the _scene*/
+	(*(_objects[2] = new DrawableObject{ _shader, 0x2f83cc_rgbf, _sphereWireframe, (*_objects[1]), _drawables }))
+		.scale(Vector3(1.01f));
+
+	(*(_objects[3] = new DrawableObject{_shader, 0xcd3431_rgbf, _plane, _scene, _drawables }))
+		.rotate(30.0_degf, Vector3(1.0f).normalized())
+		.rotateX(-30.0_degf)
+		.scale(Vector3(2.0f))
+		.translate({ 0.0f, -0.30f, -3.3f });
+
+	/* Configure camera */
+	_cameraObject = new Object3D{ &_scene };
+	_cameraObject->translate(Vector3::zAxis(8.0f));
+	_camera = new SceneGraph::Camera3D{ *_cameraObject };
+	_camera->setAspectRatioPolicy(SceneGraph::AspectRatioPolicy::Extend)
+		.setProjectionMatrix(Matrix4::perspectiveProjection(35.0_degf, 4.0f / 3.0f, 0.001f, 100.0f))
+		.setViewport(GL::defaultFramebuffer.viewport().size());
+}
+
+void PrimitivesSceneGraphExample::drawEvent() {
+    GL::defaultFramebuffer.clear(
+        GL::FramebufferClear::Color|GL::FramebufferClear::Depth);
+
+	_camera->draw(_drawables);
+    swapBuffers();
+}
+
+void PrimitivesSceneGraphExample::mousePressEvent(MouseEvent& event) {
+    if(event.button() != MouseEvent::Button::Left) return;
+
+	_previousMousePosition = _mousePressPosition = event.position();
+    event.setAccepted();
+}
+
+void PrimitivesSceneGraphExample::mouseReleaseEvent(MouseEvent& event) {	
+	if (event.button() != MouseEvent::Button::Left || _mousePressPosition != event.position()) return;
+
+	//Change the color of each object
+	for (auto* o : _objects) o->setColor(Color3::fromHsv({ o->getColor().hue() + 50.0_degf, 1.0f, 1.0f }));
+
+    event.setAccepted();
+    redraw();
+}
+
+void PrimitivesSceneGraphExample::mouseMoveEvent(MouseMoveEvent& event) {
+	/* We have to take window size, not framebuffer size, since the position is
+	in window coordinates and the two can be different on HiDPI systems */
+	const Vector2 delta = 3.0f*
+		Vector2{ event.position() - _previousMousePosition } /
+		Vector2{ windowSize() };
+
+	/* Dragging with right mouse button will rotate the camera */
+	if ((event.buttons() & MouseMoveEvent::Button::Right)) {
+		(*_cameraObject)
+			.rotate(Rad{ -delta.y() }, _cameraObject->transformation().right().normalized())
+			.rotateY(Rad{ -delta.x() });
+	}
+	
+	/* Dragging with left mouse button will rotate the objects locally */
+	if ((event.buttons() & MouseMoveEvent::Button::Left)) {
+		for (auto* o : _objects) {
+			(*o).transformLocal(Matrix4::rotationX(Rad{ delta.y() }) * Matrix4::rotationY(Rad{ delta.x() }));
```

Comment:
Would this be the same as 

```suggestion
			(*o).rotateYLocal(Rad{ delta.y() })
			    .rotateXLocal(Rad{ delta.x() }));
```

? It's a bit shorter to write.